Fix hang in SQLTransactionCoordinator::shutdown()

Ahmad-S792 opened the pull request:
#### 4a40fea04d2fc3fa045298f31baff6ca489a5959
<pre>
Fix hang in SQLTransactionCoordinator::shutdown()

<a href="https://bugs.webkit.org/show_bug.cgi?id=266552">https://bugs.webkit.org/show_bug.cgi?id=266552</a>

Reviewed by Chris Dumez.

Merge: <a href="https://src.chromium.org/viewvc/blink?view=revision&amp">https://src.chromium.org/viewvc/blink?view=revision&amp</a>;revision=181269

We should call `takeFirst` instead of `first` to decrease the size of
`info.pendingTransactions`.

Also update &apos;RefPtr&lt;SQLTransaction&gt;&apos; to &apos;RefPtr&apos;.

This changes does not have any test case.

* Source/WebCore/Modules/webdatabase/SQLTransactionCoordinator.cpp:
(SQLTransactionCoordinator::shutdown):

Canonical link: <a href="https://commits.webkit.org/272316@main">https://commits.webkit.org/272316@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ae7ee7772ab584ab683db20b7c01c3f4b3736fd5

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/31215 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/9893 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/32909 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/33722 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/28341 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/31982 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/12238 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/7151 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/27990 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/31552 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/8344 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/27909 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/7163 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/7341 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/27804 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/35063 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/28411 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/28261 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/33472 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/7386 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/5438 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/31316 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/9068 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7351 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/8086 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/7915 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/Modules/webdatabase/SQLTransactionCoordinator.cpp:
```diff
@@ -129,7 +129,7 @@ void SQLTransactionCoordinator::shutdown()
         // transaction is interrupted?" at the top of SQLTransactionBackend.cpp.
         while (!info.pendingTransactions.isEmpty()) {
-            RefPtr<SQLTransaction> transaction = info.pendingTransactions.first();
+            RefPtr<SQLTransaction> transaction = info.pendingTransactions.takeFirst();
```
Since you're modifying this line, could you please use `RefPtr` instead of `RefPtr<SQLTransaction>`.

Ahmad-S792 replied:
Done in [ae7ee77](https://github.com/WebKit/WebKit/commit/ae7ee7772ab584ab683db20b7c01c3f4b3736fd5).

Ahmad-S792 committed:
Fix hang in SQLTransactionCoordinator::shutdown()

https://bugs.webkit.org/show_bug.cgi?id=266552

Reviewed by Chris Dumez.

Merge: https://src.chromium.org/viewvc/blink?view=revision&revision=181269

We should call `takeFirst` instead of `first` to decrease the size of
`info.pendingTransactions`.

Also update 'RefPtr<SQLTransaction>' to 'RefPtr'.

This changes does not have any test case.

* Source/WebCore/Modules/webdatabase/SQLTransactionCoordinator.cpp:
(SQLTransactionCoordinator::shutdown):

Canonical link: https://commits.webkit.org/272316@main
